pkg/autodiscovery/listeners/container_test.go

```diff
@@ -274,6 +278,10 @@ func TestCreateContainerService(t *testing.T) {
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			listener, wlm := newContainerListener(t)
+
+			if tt.container != nil {
+				listener.Store().(*workloadmetatesting.Store).Set(tt.container)
+			}
 			if tt.pod != nil {
 				listener.Store().(*workloadmetatesting.Store).Set(tt.pod)
 			}
```

Comment:
There is not any chance of a conflict between the `tt.container` or the `tt.pod` value in the store here?

Reply:
Unless they have the same ID there shouldn't be any issue there. Those tests are edited manually and the unit test will fail if the ID is incorrect.